modules/core/03-connection/keeper/verify.go

```diff
@@ -28,12 +32,30 @@ func (k Keeper) VerifyClientState(
 	}
 
 	if status := targetClient.Status(ctx, clientStore, k.cdc); status != exported.Active {
-		return sdkerrors.Wrapf(clienttypes.ErrClientNotActive, "client (%s) status is %s", clientID, status)
+		return sdkerrors.Wrapf(clienttypes.ErrClientStateNotFound, "client (%s) status is %s", clientID, status)
 	}
 
-	if err := targetClient.VerifyClientState(
-		clientStore, k.cdc, height,
-		connection.GetCounterparty().GetPrefix(), connection.GetCounterparty().GetClientID(), proof, clientState); err != nil {
+	merklePath := commitmenttypes.NewMerklePath(host.FullClientStatePath(connection.GetCounterparty().GetClientID()))
+	merklePath, err := commitmenttypes.ApplyPrefix(connection.GetCounterparty().GetPrefix(), merklePath)
+	if err != nil {
+		return err
+	}
+
+	path, err := k.cdc.Marshal(&merklePath)
+	if err != nil {
+		return err
+	}
```

Comment:
I think we should remove the marshal/unmarshal and pass the merkle path type directly (that's what the spec specifies as well). We can do in a followup

Reply:
#1884